### PR TITLE
feat: convert image to grayscale

### LINKED ISSUE #287: Convert image to grayscale
### Is your feature request related to a problem?

I'd like to be able to convert an `Image` to grayscale.

### Desired solution

Add method `convert_to_grayscale` to `Image` that returns the grayscale image but leaves the original unchanged.

### Possible alternatives (optional)

_No response_

### Screenshots (optional)

_No response_

### Additional Context (optional)

_No response_

### PR DESCRIPTION
**Pull request by jxnior01**
Closes #287

### Summary of Changes

* feat: Added method that returns converted grayscale image but leaves the original unchanged.

### CODE REVIEW

File: src/safeds/data/image/containers/_image.py [hunk 1]
```diff
@@ -176,3 +176,20 @@ def resize(self, new_width: int, new_height: int) -> Image:
...
+        Convert the image to grayscale.
+
+        Returns
+        -------
+        grayscale_image : Image
+            The grayscale image.
+        """
+        data = io.BytesIO()
+        grayscale_image = self._image.convert("L")
+        grayscale_image.save(data, format=self._format.value)
```
Comment (alex-senger) [inline src/safeds/data/image/containers/_image.py:191]: Wieso wird das Bild gespeichert und nicht einfach zurückgegeben?

**Comment by lars-reimann** [comment]
## [🦙 MegaLinter](https://megalinter.io/7.0.4) status: ✅ [SUCCESS](https://github.com/Safe-DS/Stdlib/actions/runs/5290101450)

| Descriptor  |                                Linter                                 |Files|Fixed|Errors|Elapsed time|
|-------------|-----------------------------------------------------------------------|-----|----:|------|------------|
|✅ PYTHON    |[black](https://megalinter.io/7.0.4/descriptors/python_black)          |    2|    2|     0|0.86s       |
|✅ PYTHON    |[mypy](https://megalinter.io/7.0.4/descriptors/python_mypy)            |    2|     |     0|1.81s       |
|✅ PYTHON    |[ruff](https://megalinter.io/7.0.4/descriptors/python_ruff)            |    2|    2|     0|0.02s       |
|✅ REPOSITORY|[git_diff](https://megalinter.io/7.0.4/descriptors/repository_git_diff)|yes  |     |no    |0.0s        |

See detailed report in [MegaLinter reports](https://github.com/Safe-DS/Stdlib/actions/runs/5290101450)
_Set `VALIDATE_ALL_CODEBASE: true` in mega-linter.yml to validate all sources, not only the diff_

_MegaLinter is graciously provided by [OX Security](https://www.ox.security/?ref=megalinter)_

File: src/safeds/data/image/containers/_image.py [hunk 2]
```diff
@@ -221,6 +221,20 @@ def resize(self, new_width: int, new_height: int) -> Image:
...
```
Comment (alex-senger) [inline src/safeds/data/image/containers/_image.py:235]: Change the Implementation to use deepcopies.

File: src/safeds/data/image/containers/_image.py (hunk 1 above)
Comment (jxnior01) [reply src/safeds/data/image/containers/_image.py:235]: The `convert_to_grayscale()` method both saves the grayscale image in memory (without needing to convert it again) and returns an `Image` object. I saved the image to ensure that the resulting grayscale image is stored in a format that can be easily accessed and used later if needed.

**Review by alex-senger — APPROVED** [review]
[LGTM](https://lgtm.party/p/OIUJLG4O/)

**Review by alex-senger — APPROVED** [review]
[LGTM](https://lgtm.party/p/EA5K3PG/)

**Pull request merged**